index.js

```diff
@@ -83,50 +78,78 @@ Collector.prototype.computePointCountToSend = function(pointSizes, upperBound) {
   return Math.min(index, pointSizes.length); // But not if there were no points at all.
 };
 
-Collector.prototype.flush = function() {
-    var self = this;
-
-    if (!self._points || self._points.length === 0) {
+function flushSeries(collector, seriesName, points) {
+    if (!points || points.length === 0) {
         return;
     }
 
     // only send N points at a time to avoid making requests too large
-    // TODO what if we are backed up?
-
-    var spliceIndex = self.computePointCountToSend(self._points.map(function (point) {
-      return JSON.stringify(point).length;
-    }), MTU_SIZE);
-    var points = self._points.splice(0, spliceIndex);
-    var opt = { precision: self._time_precision };
+    var spliceIndex;
+    if (collector._client.protocol == 'udp:') {
+      spliceIndex = collector.computePointCountToSend(collector._points.map(function (point) {
+        return JSON.stringify(point).length;
+      }), MTU_SIZE);
+    } else {
+      spliceIndex = 50;
+    }
+    var batch = points.splice(0, spliceIndex);
+    var opt = { precision: collector._time_precision };
 
-    self._client.writePoints(self._series_name, points, opt, function(err) {
+    collector._client.writePoints(seriesName, batch, opt, function(err) {
         if (err) {
             // TODO if error put points back to send again?
             return self.emit('error', err);
```

Comment:
There's currently no `self` here. Let's make this a prototype method - this is effectively what's happening when we have the first argument be the `collector`.